### PR TITLE
Correct a H5 open issue in pims_reader.py

### PR DESCRIPTION
**Pull request by yugangzhang**
@danielballan can you take a look? Thanks!

### CODE REVIEW

File: eiger_io/fs_handler2.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            try:
+                self._entry[key]
+            except KeyError:
+                pass  # This is a link that leads nowhere.
+            else:
+                valid_keys.append(key)
+        return valid_keys
+
+    def get_frame(self, i):
+        dataset = self._entry['data_{:06d}'.format(1 + (i // self.images_per_file))]
```
Comment (tacaswell) [inline eiger_io/fs_handler2.py:35]: It might be worth caching the datasets, this `__getitem__` can be expensive.

**Comment by jrmlhermitte** [comment]
@yugangzhang do you remember how to reproduce the error?
 @jrmlhermitte  We can check some real hdf data together.

**Comment by danielballan** [comment]
Wow, I see that we're coming up on the one-year anniversary of this PR. Sorry for totally missing it, @yugangzhang.

**Comment by jrmlhermitte** [comment]
@yugangzhang what's the status? Can we close this and re-open if there are more issues?
hopefully should have been resolved by #8

**Pull request closed without merging**